src/test/java/org/prebid/server/ApplicationTest.java

```diff
@@ -467,27 +1095,88 @@ public void auctionShouldRespondWithBidsFromDifferentExchanges() throws IOExcept
+    }
+
+    @Test
+    public void auctionShouldRespondWithBidsFromRubiconAndAppnexus() throws IOException {
+        // given
+        // rubicon bid response for ad unit 1
+        wireMockRule.stubFor(post(urlPathEqualTo("/rubicon-exchange"))
+                .withQueryParam("tk_xint", equalTo("rp-pbs"))
+                .withBasicAuth("rubicon_user", "rubicon_password")
+                .withHeader("Content-Type", equalToIgnoreCase("application/json;charset=utf-8"))
+                .withHeader("Accept", equalTo("application/json"))
+                .withHeader("User-Agent", equalTo("prebid-server/1.0"))
+                .withRequestBody(equalToJson(jsonFrom("auction/rubicon&appnexus/test-rubicon-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/rubicon&appnexus/test-rubicon-bid-response-1.json"))));
+
+        // rubicon bid response for ad unit 2
+        wireMockRule.stubFor(post(urlPathEqualTo("/rubicon-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("auction/rubicon&appnexus/test-rubicon-bid-request-2.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/rubicon&appnexus/test-rubicon-bid-response-2.json"))));
+
+        // rubicon bid response for ad unit 3
+        wireMockRule.stubFor(post(urlPathEqualTo("/rubicon-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("auction/rubicon&appnexus/test-rubicon-bid-request-3.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/rubicon&appnexus/test-rubicon-bid-response-3.json"))));
+
+        // appnexus bid response for ad unit 4
+        wireMockRule.stubFor(post(urlPathEqualTo("/appnexus-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("auction/rubicon&appnexus/test-appnexus-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/rubicon&appnexus/test-appnexus-bid-response-1.json"))));
+
+        // pre-bid cache
+        wireMockRule.stubFor(post(urlPathEqualTo("/cache"))//
+                .withRequestBody(equalToJson(jsonFrom("auction/rubicon&appnexus/test-cache-rubicon-appnexus-request.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/rubicon&appnexus/test-cache-rubicon-appnexus-response.json"))));
+
+        final Map<String, String> exchanges = new HashMap<>();
+        exchanges.put(RUBICON, "http://localhost:" + WIREMOCK_PORT + "/rubicon-exchange?tk_xint=rp-pbs");
+        exchanges.put(APPNEXUS, "http://localhost:" + WIREMOCK_PORT + "/appnexus-exchange");
+        exchanges.put(APPNEXUS_ALIAS, null);
 
         // when
         final Response response = given(spec)
                 .header("Referer", "http://www.example.com")
                 .header("X-Forwarded-For", "192.168.244.1")
                 .header("User-Agent", "userAgent")
                 .header("Origin", "http://www.example.com")
-                // this uids cookie value stands for
```

Comment:
I would update this comment and keep it as it's effectively decrypting following cookie value